ExpandDims doesn't accept negative indices anymore

#### Issue Description

By fixing issue #9847, you have effectively prevented the use of negative indexes in `expandDims`, i.e., the following no longer works:
```
INDArray v1 = Nd4j.ones(2, 2);
System.out.println(Arrays.toString(v1.shape())); // [2, 2]
INDArray v2 = Nd4j.expandDims(v1, -1); // throws exception
System.out.println(Arrays.toString(v2.shape())); // should now be [2, 2, 1]
```

#### Version Information

* Deeplearning4j version: 1.0.0-SNAPSHOT
* Platform information (OS, etc): Linux Mint 21
* CUDA version, if used: N/A
* NVIDIA driver version, if in use: N/A

#### Additional Information

* Exception stack trace:
```
Error at [/home/runner/work/deeplearning4j/deeplearning4j/libnd4j/include/ops/declarable/generic/shape/expand_dims.cpp:73:0]:
ExpandDims: axis should be in range of 0...3 in this case, but got -1 instead
[main] ERROR org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner - Failed to calculate output shapes for op expand_dims. Attempted to execute with 1 inputs, 0 outputs, 0 targs, 1 iargs, 0 bargs and 0 dargs. Inputs: [(FLOAT,[1,1],c)] - Please see above message (printed out from c++) for a possible cause of error.
Exception in thread "main" org.nd4j.linalg.exception.ND4JIllegalStateException: Op name expand_dims - no output arrays were provided and calculateOutputShape failed to execute
	at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.exec(NativeOpExecutioner.java:1360)
	at org.nd4j.linalg.factory.Nd4j.exec(Nd4j.java:6575)
	at org.nd4j.linalg.factory.ops.NDBase.expandDims(NDBase.java:520)
	at org.nd4j.linalg.factory.Nd4j.expandDims(Nd4j.java:416)
	at ExpandDims.main(ExpandDims.java:18)
Caused by: java.lang.RuntimeException: Op expand_dims with name null failed to execute. Here is the error from c++: Op validation failed
	at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.calculateOutputShape(NativeOpExecutioner.java:1501)
	at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.calculateOutputShape(NativeOpExecutioner.java:1407)
	at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.exec(NativeOpExecutioner.java:1349)
	... 5 more
```

--- a/libnd4j/include/ops/declarable/generic/shape/expand_dims.cpp
+++ b/libnd4j/include/ops/declarable/generic/shape/expand_dims.cpp
@@ -68,11 +68,12 @@ DECLARE_SHAPE_FN(expand_dims) {
   char order = shape::order(inShape);
 
   sd::LongType axis = block.numI() > 0 ? INT_ARG(0) : INPUT_VARIABLE(1)->e<int>(0);
+  if (axis < 0) axis += x_rank + 1;
+
   REQUIRE_TRUE(axis >= 0 && axis <= input->rankOf(), 0,
                "ExpandDims: axis should be in range of 0...%i in this case, but got %i instead", input->rankOf() + 1,
                axis);
 
-  if (axis < 0) axis += x_rank + 1;
   std::vector<sd::LongType> shape;
   for (int e = 0; e < x_rank; e++) shape.emplace_back(shape::shapeOf(inShape)[e]);
 

--- a/platform-tests/src/test/java/org/eclipse/deeplearning4j/nd4j/autodiff/opvalidation/TestShapeOpValidation.java
+++ b/platform-tests/src/test/java/org/eclipse/deeplearning4j/nd4j/autodiff/opvalidation/TestShapeOpValidation.java
@@ -235,11 +235,19 @@ public class TestShapeOpValidation extends BaseOpValidation {
     @MethodSource("org.nd4j.linalg.BaseNd4jTestWithBackends#configs")
     @ParameterizedTest()
     public void testExpandDimsOutofBounds(Nd4jBackend backend) {
-       assertThrows(ND4JIllegalStateException.class,() -> {
-           INDArray v1 = Nd4j.zeros(1, 1);
-           INDArray v2 = Nd4j.base().expandDims(v1, 3); // crashes
-       });
+        assertThrows(ND4JIllegalStateException.class,() -> {
+            INDArray v1 = Nd4j.zeros(1, 1);
+            INDArray v2 = Nd4j.base().expandDims(v1, 3); // crashes
+        });
 
+    }
+
+    @ParameterizedTest
+    @MethodSource("org.nd4j.linalg.BaseNd4jTestWithBackends#configs")
+    public void testNegativeIndicesExpandDims(Nd4jBackend backend) {
+        INDArray v1 = Nd4j.ones(2, 2);
+        INDArray v2 = Nd4j.expandDims(v1, -1); // throws exception
+        assertArrayEquals(new long[]{2,2,1},v2.shape());
     }
 
     @MethodSource("org.nd4j.linalg.BaseNd4jTestWithBackends#configs")

